src/main/java/uk/ac/cam/cl/dtg/segue/api/GroupsFacade.java

```diff
@@ -471,7 +448,103 @@ public Response deleteGroup(@Context final HttpServletRequest request,
         } catch (NoUserLoggedInException e) {
             return SegueErrorResponse.getNotLoggedInResponse();
         }
-        
+
         return Response.noContent().build();
     }
-}
+
+    @POST
+    @Path("{group_id}/manager")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response addAdditionalManagerToGroup(@Context final HttpServletRequest request,
+                                                @PathParam("group_id") final Long groupId,
+                                                final Map<String, String> responseMap) {
+        if (null == groupId) {
+            return new SegueErrorResponse(Status.BAD_REQUEST, "Group must be specified.").toResponse();
+        }
+
+        if (null == responseMap && responseMap.containsKey("email") && !responseMap.get("email").isEmpty()) {
+            return new SegueErrorResponse(Status.BAD_REQUEST, "User email must be specified.").toResponse();
+        }
+
+        try {
+            RegisteredUserDTO user = userManager.getCurrentRegisteredUser(request);
+            RegisteredUserDTO userToAdd = this.userManager.getUserDTOByEmail(responseMap.get("email"));
+
+            if (null == userToAdd || Role.STUDENT.equals(userToAdd.getRole())) {
+                // TODO: maybe add misuse monitor to this endpoint
+                // deliberately be vague about whether the account exists or they don't have a teacher account to avoid account scanning.
+                return new SegueErrorResponse(Status.BAD_REQUEST, "There was a problem adding the user specified. Please make sure their email address is correct and they have a teacher account.").toResponse();
+            }
+
+            UserGroupDTO group = groupManager.getGroupById(groupId);
+
+            if(group.getOwnerId().equals(userToAdd.getId())) {
+                return new SegueErrorResponse(Status.BAD_REQUEST, "The owner cannot be added as an additional manager").toResponse();
+            }
+
+            boolean idExists = GroupManager.isInAdditionalManagerList(group, userToAdd.getId());
+
+            if(idExists) {
+                return new SegueErrorResponse(Status.BAD_REQUEST, "This user already exists as an additional manager").toResponse();
+            }
+
+            if (!group.getOwnerId().equals(user.getId()) && !isUserAnAdmin(userManager, request)) {
+                return new SegueErrorResponse(Status.FORBIDDEN, "Only group owners or admins modify groups managers").toResponse();
+            }
+
+            this.getLogManager().logEvent(user, request, ADD_ADDITIONAL_GROUP_MANAGER,
```

Comment:
These log messages occur before the action has actually taken place. I think this may happen elsewhere too, but it means that a database error will leave the logged_events table in a state inconsistent with the actual state.